### PR TITLE
fix: pull from al2 from ecr

### PR DESCRIPTION
**Pull request by guoqiaoli1992**
Issue #, if available:
Reaching to docker limit, need to pull al2 from ecr repo

Description of changes:
login to ecr repo account first and change "FROM amazonlinux:2" to "FROM 137112412989.dkr.ecr.us-east-1.amazonaws.com/amazonlinux:2"

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -22,10 +22,14 @@ source scripts/shared.sh
 
 parse_std_args "$@"
 
+aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin 137112412989.dkr.ecr.us-west-2.amazonaws.com
```
Comment (andremoeller) [inline scripts/build.sh:25]: (1) Does this work in non `us-west-2` regions? (Note there's `--region us-west-2`, as well as `us-west-2` in the image URI

(2) Could you add a link to this docs page: https://docs.aws.amazon.com/AmazonECR/latest/userguide/amazon_linux_container_image.html

File: scripts/build.sh [hunk 2]
```diff
@@ -22,10 +22,14 @@ source scripts/shared.sh
...
+
 echo "building image ${version} ... "
 docker build \
     -f ${build_context}/docker/Dockerfile.${processor} \
     -t ${repository}:${version} \
     --build-arg REGION=${REGION} \
     -t sagemaker-spark:latest \
     ${build_context}
+
+docker logout https://137112412989.dkr.ecr.us-west-2.amazonaws.com
```
Comment (andremoeller) [inline scripts/build.sh:35]: I recommend putting this URI, which is the same as the Image URI, in a bash variable.

File: src/smspark/job.py [hunk 3]
```diff
@@ -32,6 +32,9 @@
 class ProcessingJobManager(object):
     """Manages the lifecycle of a Spark job."""
 
+    _bootstrapping_timeout = 600.0  # all hosts should report as ready within this timeout.
+    _wait_for_primary_timeout = 600.0  # then, all workers ask the primary if it's up within this timeout.
+
```
Comment (andremoeller) [inline src/smspark/job.py:37]: Could you track a follow-up to merge this into the `master` branch too? Just to make sure they don't end up getting out of sync.

**Review by andremoeller — CHANGES_REQUESTED** [review]

File: src/smspark/job.py (hunk 3 above)
Comment (guoqiaoli1992) [reply src/smspark/job.py:37]: sure, eventually all branches will be merged into mainline, including this one.

File: scripts/build.sh (hunk 1 above)
Comment (guoqiaoli1992) [reply scripts/build.sh:25]: 1) yeah because we only build in alpha, and other stages/regions will just pull image from alpha

2) Sure.

**Review by andremoeller — APPROVED** [review]

**Pull request merged**